File: 1143.cpp
class Solution {
public:
 int longestCommonSubsequence(string s, string ss) {
	int sizes = s.size(); 
	int sizess = ss.size(); 
	int b = 0;
	int m[sizes+1][sizess+1]; 
	memset(m, 0 , sizeof(m)); 
	for (int i = 1;i <=  sizes; i++) {
		for (int j = 1; j <= sizess; j++){
			if (s[i-1] == ss[j-1])
				m[i][j] = m[i-1][j-1]+1; 
			else{
				m[i][j] = max(m[i-1][j], m[i][j-1]); 
			}
		   b = m[i][j] > b? m[i][j]: b; 	
		} 

	} 
	return b ; 
        
 }
};

File: 1143.java
class Solution {
    public int longestCommonSubsequence(String t1, String t2) {
        int s1 = t1.length(), s2 = t2.length(); 
        int[][] m = new int[s1+1][s2+1];
            
        for (int i = 0 ; i < s1; i++){
            for(int j = 0 ; j < s2; j++){
                m[i+1][j+1] = t1.charAt(i) != t2.charAt(j) ? Math.max(m[i][j+1], m[i+1][j]): m[i][j]+1; 
            }
        }
        
        return m[s1][s2]; 
    }
    
}
